### PR TITLE
[WIP] Added syntax keyword item spec ui test

### PR DESCRIPTION
**Pull request by geonik**
Hi guys,

I made this simple ui test just to get my feet wet with neovim.

I was thinking of going ahead and writing more ui tests relevant to the :syntax command.

Will this be useful for the project or would it be better to focus on another area?

### CODE REVIEW

**Comment by cacplate** [comment]
You could refer to #1630 for every test you make like this. Alternatively, other than writing more tests, converting legacy tests can be something to do (not that it's more important but if you ever search what to do).

**Comment by geonik** [comment]
Thanks for the tip.

I will go on with the syntax command for the time beeing.

As I read #1630 I think these kinds of tests are relevant to the issue.

**Comment by geonik** [comment]
I checked the error shown by the CI and it seems unrelated to the PR. It seems that a download of llvm with curl fails and this causes the job to fail.

File: test/functional/ui/syntax_keyword_item_spec.lua [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+local eq = helpers.eq
+local eval = helpers.eval
+
+describe('Syntax', function()
+  local screen
+  local colors = Screen.colors
+
+  before_each(function()
+    clear()
+    screen = Screen.new()
```
Comment (justinmk) [inline test/functional/ui/syntax_keyword_item_spec.lua]: consider `Screen.new(25, 5)` or something like that for a smaller screen. See e.g. `mouse_spec.lua`. Unless you plan on needing the space for the tests, it's nice to condense the test files to make them easier to read.

**Comment by justinmk** [comment]
> As I read #1630 I think these kinds of tests are relevant to the issue.

Yes. Though, the particular tests in this PR are sufficiently covered by `functional/ui/searchhl_spec.lua`. (I found this via `git grep syntax -- test/`). But I don't see many syntax tests in the legacy `src/nvim/testdir` (you might double-check that), so other more specific tests would be useful.

Thanks!

**Comment by geonik** [comment]
> Yes. Though, the two tests in this PR at present are sufficiently covered by functional/ui/searchhl_spec.lua

I did see that but my intention is to write a spec that covers the full behaviour of the syntax command as specified in the syntax.txt helpfile. I know that this will replicate some tests in searchhl_spec.lua but I thought for the sake of completeness to include it in this test also.

I am changing the tag to [WIP] since I will be adding commits until I have the syntax command covered fully.

@justinmk thanks for the feedback

**Comment by geonik** [comment]
Closing since #1630 has been also closed.

Will look around for some other item to contribute.

**Comment by justinmk** [comment]
@geonik If you added _new_ tests please keep this open, I'll look at it. I didn't notice your new commits (GitHub does not notify about this, please leave a comment instead)

**Comment by justinmk** [comment]
Doesn't appear to add new tests, so closing.

**Pull request closed without merging**